### PR TITLE
Fixes #6 -- expose if an EOF happens in Decoder.Decoder

### LINKED ISSUE #6: No way to tell if at EOF when using Decoder.Decode() with struct
Looking at:

https://github.com/ianlopshire/go-fixedwidth/blob/master/decode.go#L81-L93

and

https://github.com/ianlopshire/go-fixedwidth/blob/master/decode.go#L113-L128

As far as I can tell, `readLine` eats the EOF error -- it returns an `ok` bool, but `Decode` throws that away. This could be fixed by having `Decode` transform the `ok` bool back into an `err == io.EOF`, or the `done` bool could be exposed via a public API.

I'm happy to implement a PR for either one of these strategies that you prefer.

Thanks for this package!

### PR DESCRIPTION
**Pull request by alex**
You'll see that there were two tests I needed to change. The first change I think is definitely right -- you can't read a struct with fields from empty data. The second one I'm less confident in (see the TODO comment I left), I'd appreciate your feedback on that.

### CODE REVIEW

File: decode.go [hunk 1]
```diff
@@ -88,7 +89,10 @@ func (d *Decoder) Decode(v interface{}) error {
 		return d.readLines(reflect.ValueOf(v).Elem())
 	}
 
-	err, _ := d.readLine(reflect.ValueOf(v))
+	err, ok := d.readLine(reflect.ValueOf(v))
+	if d.done && err == nil && !ok {
```
Comment (ianlopshire) [inline decode.go]: @alex can you add a comment here to explain why `io.EOF` is returned?

File: decode_test.go [hunk 2]
```diff
@@ -219,3 +223,30 @@ func TestNewValueSetter(t *testing.T) {
 		})
 	}
 }
+
+func TestDecode(t *testing.T) {
```
Comment (ianlopshire) [inline decode_test.go]: @alex Can you rename this test `TestDecode_EOF` and add a comment that links back to the issue.

File: decode_test.go [hunk 3]
```diff
@@ -149,7 +151,9 @@ func TestUnmarshal(t *testing.T) {
 			{"Invalid Unmarshal Not Pointer 1", struct{}{}, true},
 			{"Invalid Unmarshal Not Pointer 2", []struct{}{}, true},
 			{"Valid Unmarshal slice", &[]struct{}{}, false},
-			{"Valid Unmarshal struct", &struct{}{}, false},
+			// TODO: technically you can unmarshal an empty struct from empty
```
Comment (ianlopshire) [inline decode_test.go]: I think this should throw an EOF.

**Comment by ianlopshire** [comment]
@alex thanks for submitting this PR! Everything looks great, just a couple of nitpicky requests.

**Comment by alex** [comment]
Got 'em! Thanks for the quick review!

**Comment by ianlopshire** [comment]
@alex Thanks! merged!

**Comment by alex** [comment]
Thank you!

**Pull request merged**